libbz2-rs-sys/src/blocksort.rs

```diff
@@ -595,28 +595,24 @@ fn mainQSort3(
     let mut n: i32;
     let mut m: i32;
     let mut med: i32;
-    let mut sp: i32;
     let mut lo: i32;
     let mut hi: i32;
     let mut d: i32;
-    let mut stackLo: [i32; 100] = [0; 100];
-    let mut stackHi: [i32; 100] = [0; 100];
-    let mut stackD: [i32; 100] = [0; 100];
+
+    let mut stack = [(0i32, 0i32, 0i32); 100];
+
     let mut nextLo: [i32; 3] = [0; 3];
     let mut nextHi: [i32; 3] = [0; 3];
     let mut nextD: [i32; 3] = [0; 3];
-    sp = 0;
-    stackLo[sp as usize] = loSt;
-    stackHi[sp as usize] = hiSt;
-    stackD[sp as usize] = dSt;
-    sp += 1;
+
+    stack[0] = (loSt, hiSt, dSt);
+
+    let mut sp = 1;
     while sp > 0 {
-        assert_h!(sp < MAIN_QSORT_STACK_SIZE - 2, 1001);
+        assert_h!(sp < MAIN_QSORT_STACK_SIZE as usize - 2, 1001);
 
         sp -= 1;
-        lo = stackLo[sp as usize];
-        hi = stackHi[sp as usize];
-        d = stackD[sp as usize];
+        (lo, hi, d) = stack[sp];
```

Comment:
```suggestion
        let (lo, hi, d) = stack[sp];
```

to make it clear that these variables are only used within a single loop iter.